File: Tutorial4-8LayerLibrary/src/main/java/com/test/tutorial4_8layerlibrary/BaseApplication.java
package com.test.tutorial4_8layerlibrary;

import java.util.Random;

import dagger.android.support.DaggerApplication;

public abstract class BaseApplication extends DaggerApplication {


    @Override
    public void onCreate() {
        super.onCreate();
    }


    public int testResult() {
        Random random = new Random();
        return random.nextInt(100);
    }

}

File: Tutorial4-8LayerLibrary/src/main/java/com/test/tutorial4_8layerlibrary/BaseApplication.kt
package com.test.tutorial4_8layerlibrary

import dagger.android.support.DaggerApplication
import java.util.*

abstract class BaseApplication : DaggerApplication() {

    override fun onCreate() {
        super.onCreate()
    }

    fun testResult(): Int {
        val random = Random()
        return random.nextInt(100)
    }

}